pymc/distributions/timeseries.py

```diff
@@ -33,6 +35,159 @@
+
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        mu: Union[np.ndarray, float],
+        sigma: Union[np.ndarray, float],
+        init: float,
+        steps: int,
+        size: Tuple[int],
+    ) -> np.ndarray:
+        """Gaussian Random Walk generator.
+
+        The init value is drawn from the Normal distribution with the same sigma as the
+        innovations.
+
+        Notes
+        -----
+        Currently does not support custom init distribution
+
+        Parameters
+        ----------
+        rng: np.random.RandomState
+           Numpy random number generator
+        mu: np.ndarray
+           Random walk mean
+        sigma: np.ndarray
+            Standard deviation of innovation (sigma > 0)
+        init: float
+            Initialization value for GaussianRandomWalk
+        steps: int
+            Length of random walk, must be greater than 1. Returned array will be of size+1 to
+            account as first value is initial value
+        size: int
+            The number of Random Walk time series generated
+
+        Returns
+        -------
+        np.ndarray
+        """
+
+        if steps is None or steps < 1:
+            raise ValueError("Steps must be None or greater than 0")
+
+        # If size is None then the returned series should be (1+steps,)
+        if size is None:
+            init_size = 1
+            steps_size = steps
+
+        # TODO: Ask Ricardo how size is known to be a tuple? Its int above
+        # If size is None then the returned series should be (size, 1+steps)
+        else:
+            init_size = (*size, 1)
+            steps_size = (*size, steps)
+
+        init_val = rng.normal(init, sigma, size=init_size)
+        steps = rng.normal(loc=mu, scale=sigma, size=steps_size)
+
+        grw = np.concatenate([init_val, steps], axis=-1)
+
+        return np.cumsum(grw, axis=-1)
+
+
+gaussianrandomwalk = GaussianRandomWalkRV()
+
+
+class GaussianRandomWalk(distribution.Continuous):
+    r"""Random Walk with Normal innovations
+
+
+    Notes
+    -----
+    init is currently drawn from a Normal distribution with the same sigma as the innovations
+
+    Parameters
+    ----------
+    mu: tensor
+        innovation drift, defaults to 0.0
+    sigma: tensor
+        sigma > 0, innovation standard deviation, defaults to 0.0
+    init: float
+        Mean value of initialization, defaults to 0.0
+    steps: int
+        Number of steps in Gaussian Random Walks
+    size: int
+        Number of independent Gaussian Random Walks
+    """
+
+    rv_op = gaussianrandomwalk
+
+    @classmethod
+    def dist(
+        cls,
+        mu: Optional[Union[np.ndarray, float]] = 0.0,
+        sigma: Optional[Union[np.ndarray, float]] = 1.0,
+        init: float = 0.0,
+        steps: int = 0,
+        *args,
+        **kwargs
+    ) -> RandomVariable:
+
+        params = [at.as_tensor_variable(param) for param in (mu, sigma, init, steps)]
```

Comment:
We usually wrap parameters in either `floatX` or `intX`. So in this case something like:
```suggestion
        params = [at.as_tensor_variable(floatX(param)) for param in (mu, sigma, init)] + [at.as_tensor_variable(intX(steps))]
```